eskip/eskip.go

```diff
@@ -44,13 +53,28 @@ type parsedRoute struct {
 	backend  string
 }
 
+// A CustomPredicate object represents a parsed, in-memory, route matching predicate
+// that is defined by extensions.
+type CustomPredicate struct {
```

Comment:
so.. now i reduced a bit the usage of the term 'Custom', because what is custom. On the other hand, while I it is true that it was a bit short-sighted back then to hardcode the types of the prepacked predicates, and it would be much nicer to just provide a built-in set of commonly used predicate implementations, this change can easily become an epic refactor, also hard to do it in a backwards compatible way, so I think we should do it in a separate PR. Can't wait for it though :)